mono/metadata/class-internals.h

```diff
@@ -1257,7 +1257,7 @@ mono_class_get_checked (MonoImage *image, guint32 type_token, MonoError *error);
 MonoClass *
 mono_class_get_and_inflate_typespec_checked (MonoImage *image, guint32 type_token, MonoGenericContext *context, MonoError *error);
 
-MonoClass *
+G_EXTERN_C MonoClass* // FIXME? G_EXTERN_C due to MonoError varying
```

Comment:
Confused why _this_ function has to be `G_EXTERN_C` but none of the others functions in this header do too.

Reply:
I not sure -- "because there was an error in CI and I was lazy about it"?
Maybe "because profiler not C++"? I can undo it and get crispness on it.
I think it was driver and dis though, which I think are C++, so that does not hold.
I can revisit, before or after this PR, as you wish.

Reply:
Tests do not include config.h.
So they don't have ifdef inside_runtime.
And they test external_only, which including config.h would break (deprecation error).

Solutions:

1. extern "C" as I had, but with comments explaining.

*or*

2. include config.h and muck with external_only

*or*

3. Still do not include config.h, define inside_runtime, muck with external_only

*or*

4. Another define to guide the internals, like, `MONO_ALLOW_EXTERNAL_ONLY`.

I pushed #3.

Reply:
https://github.com/mono/mono/pull/16218/commits/23a75a132bc2dd2b9642eb30f70f4fc80fabeeb2